### PR TITLE
 Procedure operations to use v2 APIs

### PR DESCRIPTION
**Pull request by anwar6953**
V3 procedure APIs will be removed.
In order to facilitate that, moving procedure operations from CLI + Client to hit the V2 endpoints.
These effective changes can be removed once procedures are completely removed from CDAP.

Now, from the CLI, procedure operations will only function if the user is in the default namespace. Otherwise, they will receive an error message:
```Error: Procedure operations are only supported in the default namespace.```

Build: http://builds.cask.co/browse/CDAP-DUT856-4

### CODE REVIEW

File: cdap-client/src/main/java/co/cask/cdap/client/util/VersionMigrationUtils.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import co.cask.cdap.proto.ProgramType;
+import com.google.common.base.Preconditions;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/**
+ * Utility methods used by CDAP Clients for CDAP API version migration
+ */
+public class VersionMigrationUtils {
```
Comment (bdmogal) [inline cdap-client/src/main/java/co/cask/cdap/client/util/VersionMigrationUtils.java]: Should these methods be in ``RESTMigrationUtils`` in ``cdap-common``. Don't feel too strongly, but would like to keep static helper classes to the minimum.

Comment (anwar6953) [inline cdap-client/src/main/java/co/cask/cdap/client/util/VersionMigrationUtils.java]: The methods exposed are very cdap-client specific (dealing with ClientConfig and resolveURL that uses clientConfig), so I think its fine to keep it here.
Will be gone 3.0.

File: cdap-client/src/main/java/co/cask/cdap/client/util/VersionMigrationUtils.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+   */
+  public static boolean isProgramSupported(ClientConfig config, ProgramType programType) {
+    return !(ProgramType.PROCEDURE == programType) || isProcedureSupported(config);
+  }
+
+  /**
+   * @throws IllegalStateException if procedures are not supported with the given configuration
+   * @param config the config object being used by a client
+   */
+  public static void assertProcedureSupported(ClientConfig config) {
```
Comment (bdmogal) [inline cdap-client/src/main/java/co/cask/cdap/client/util/VersionMigrationUtils.java]: Perhaps this doesn't need to be a method on its own? Or are you doing it so you do not have to repeat the error message at multiple places?

Comment (bdmogal) [inline cdap-client/src/main/java/co/cask/cdap/client/util/VersionMigrationUtils.java]: Actually ignore this comment. I see why you would need it.

File: cdap-client/src/main/java/co/cask/cdap/client/util/VersionMigrationUtils.java (hunk 1 above)
Comment (bdmogal) [inline cdap-client/src/main/java/co/cask/cdap/client/util/VersionMigrationUtils.java]: ``ClientConfig`` won't be available in ``RESTMigrationUtils``, right? Makes sense then.

**Comment by bdmogal** [comment]
LGTM, pending build.

**Comment by jwang47** [comment]
LGTM

**Pull request merged**